hibernate-reactive-core/src/main/java/org/hibernate/reactive/mutiny/impl/MutinySessionFactoryImpl.java

```diff
@@ -116,6 +123,13 @@ private SessionCreationOptions options(String tenantIdentifier) {
 				.tenantIdentifier( tenantIdentifier );
 	}
 
+	Uni<Mutiny.StatelessSession> newStatelessSession(String tenantId) {
+		return uni( () -> connection( tenantId ) )
+				.map( reactiveConnection -> new ReactiveStatelessSessionImpl(
```

Comment:
this needs to change to 
```
.chain( reactiveConnection -> create( reactiveConnection, () -> new ReactiveStatelessSessionImpl( delegate, options( tenantId ), reactiveConnection ) ) )
```

See the other similar methods for reference.
Otherwise the connection doesn't get closed if an error occurs in the constructor of the session